BLE_GattClient/source/BLEProcess.h

```diff
@@ -139,91 +146,104 @@ class BLEProcess : private mbed::NonCopyable<BLEProcess> {
 
     /**
      * Start the gatt client process when a connection event is received.
+     * This is called by Gap to notify the application we connected
      */
-    void when_connection(const Gap::ConnectionCallbackParams_t *connection_event)
-    {
-        printf("Connected.\r\n");
+    virtual void onConnectionComplete(
+        const ble::ConnectionCompleteEvent &event
+    ) {
+        if (event.getStatus() == BLE_ERROR_NONE) {
+            printf("Connected.\r\n");
+        } else {
+            printf("Failed to connect\r\n");
+            _event_queue.call(this, &BLEProcess::start_advertising);
+        }
     }
 
     /**
      * Stop the gatt client process when the device is disconnected then restart
      * advertising.
+     * This is called by Gap to notify the application we disconnected
      */
-    void when_disconnection(const Gap::DisconnectionCallbackParams_t *event)
-    {
+    virtual void onDisconnectionComplete(
+        const ble::DisconnectionCompleteEvent &event
+    ) {
         printf("Disconnected.\r\n");
-        start_advertising();
-    }
-
-    /**
-     * Setup the advertising parameters.
-     */
-    void setup_advertising()
-    {
-        Gap &gap = _ble_interface.gap();
-        gap.setAdvertisingPayload(make_advertising_data());
-        gap.setAdvertisingParams(make_advertising_params());
+        _event_queue.call(this, &BLEProcess::start_advertising);
     }
 
     /**
      * Start the advertising process; it ends when a device connects.
      */
     void start_advertising()
     {
-        ble_error_t error = _ble_interface.gap().startAdvertising();
-        if (error) {
-            printf("Error %u during gap.startAdvertising.\r\n", error);
-        } else {
-            printf("Advertising started.\r\n");
-        }
-    }
+        ble_error_t error;
 
-    /**
-     * Schedule processing of events from the BLE middleware in the event queue.
-     */
-    void schedule_ble_events(BLE::OnEventsToProcessCallbackContext *event)
-    {
-        _event_queue.call(mbed::callback(&event->ble, &BLE::processEvents));
-    }
+        ble::AdvertisingParameters adv_params;
+        adv_params.setUseLegacyPDU(false);
```

Comment:
We do want to use the legacy advertising as it works with all the BLE devices out there.

Reply:
[Done](https://github.com/ARMmbed/mbed-os-example-ble/compare/4b8ee9cd644d5315d74602f636c4d6f44a29cd2e..3f15044595ebaf5a90644c562cc2904d2087c8f6).